src/dhtproto/client/request/internal/GetChannels.d

```diff
@@ -0,0 +1,367 @@
+            (error status)
+
+    ***************************************************************************/
+
+    private bool handleStatusCode ( ubyte status )
+    {
+        auto getchannels_status = cast(RequestStatusCode)status;
+
+        if ( GetChannels.handleGlobalStatusCodes(getchannels_status,
+            this.context, this.conn.remote_address) )
+        {
+            return false; // Global code, e.g. request/version not supported
+        }
+
+        // GetChannels-specific codes
+        with ( RequestStatusCode ) switch ( getchannels_status )
+        {
+            case Started:
+                // Expected "request started" code
+                return true;
+
+            case Error:
+                // The node returned an error code. Notify the user and
+                // end the request.
+                GetChannels.Notification n;
+                n.node_error = RequestNodeInfo(
+                    this.context.request_id, conn.remote_address);
+                GetChannels.notify(this.context.user_params, n);
+                return false;
+
+            default:
+                // Treat unknown codes as internal errors.
+                goto case Error;
+        }
+
+        assert(false);
+    }
+
+    /***************************************************************************
+
+        Handler policy, called from AllNodesRequest template to run the
+        request's main handling logic.
+
+    ***************************************************************************/
+
+    private void handle ( )
+    {
+        // Receive the incoming record stream.
+        bool finished;
+        do
+        {
+            this.conn.receive(
+                ( in void[] payload )
+                {
+                    finished = this.handleMessage(payload);
+                }
+            );
+        }
+        while ( !finished );
+    }
+
+    /***************************************************************************
+
+        Handles a single message received from the node.
+
+        Params:
+            payload = message payload
+
+        Returns:
+            true if the message indicates that the request is finished
+
+    ***************************************************************************/
+
+    private bool handleMessage ( Const!(void)[] payload )
+    {
+        bool finished;
+
+        auto msg_type =
+            *this.conn.message_parser.getValue!(MessageType)(payload);
+        with ( MessageType ) switch ( msg_type )
+        {
+            case ChannelName:
+                Const!(void)[] channel;
+                this.conn.message_parser.parseBody(payload, channel);
+                Notification notification;
+                notification.received =
+                    RequestDataInfo(context.request_id, channel);
+                GetChannels.notify(this.context.user_params, notification);
+                break;
+
+            case Finished:
+                finished = true;
```

Comment:
Here just `return true` and `return false` at the end of the method? Saves you a boolean variable. Maybe current approach is cleaner.

Reply:
No I think a direct `return` is cleaner.